### PR TITLE
[ticket/14970] Use stylelint.io in travis CI builds

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14970

### CODE REVIEW

**Comment by marc1706** [comment]
Seems to run now. Thoughts @hanakin ?

**Comment by hanakin** [comment]
yeah there are going to be some changes that still need to happen to the stylelint config but should not effect this. It will provide a good test though to make sure its working

**Comment by marc1706** [comment]
Error / warning thrown by stylelint causes tests to fail as expected: https://travis-ci.org/phpbb/phpbb/jobs/190053358

File: travis/check-stylesheet.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# This file is part of the phpBB Forum Software package.
+#
+# @copyright (c) phpBB Limited <https://www.phpbb.com>
+# @license GNU General Public License, version 2 (GPL-2.0)
+#
+# For full copyright and license information, please see
+# the docs/CREDITS.txt file.
+#
+set -e
+set -x
```
Comment (Nicofuma) [inline travis/check-stylesheet.sh]: removes it please, or add set +x just before sourcing nvm because the output is 1000 times too verbose

Comment (marc1706) [inline travis/check-stylesheet.sh]: @Nicofuma better?

**Comment by marc1706** [comment]
Builds still fail on incorrect input data: https://travis-ci.org/phpbb/phpbb/jobs/190623874

File: phpBB/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "author": "",
+  "license": "GPL-2.0",
+  "bugs": {
+    "url": "https://github.com/phpbb/phpbb/issues"
+  },
+  "homepage": "https://github.com/phpbb/phpbb#readme",
+  "devDependencies": {
+    "stylelint": "^7.7.1"
+  }
```
Comment (hanakin) [inline phpBB/package.json]: need to add new dependency for future versions `"stylelint-order": "^0.3.0"`

File: phpBB/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "phpbb",
+    "forum",
+    "php",
+    "software",
+    "community"
+  ],
+  "author": "",
+  "license": "GPL-2.0",
+  "bugs": {
+    "url": "https://github.com/phpbb/phpbb/issues"
```
Comment (hanakin) [inline phpBB/package.json]: this should technically point to the tracker since we do not use issues

File: phpBB/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  "homepage": "https://github.com/phpbb/phpbb#readme",
```
Comment (hanakin) [inline phpBB/package.json]: this should probably point to .com

File: travis/check-stylesheet.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# @license GNU General Public License, version 2 (GPL-2.0)
+#
+# For full copyright and license information, please see
+# the docs/CREDITS.txt file.
+#
+set -e
+set +x
+
+DB=$1
+TRAVIS_PHP_VERSION=$2
```
Comment (VSEphpbb) [inline travis/check-stylesheet.sh]: I don't see any need for DB or TRAVIS_PHP_VERSION

**Comment by marc1706** [comment]
Updated

**Comment by hanakin** [comment]
👍

File: travis/check-stylesheet.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	# Note that you can just hardcode a preferred version here.
+	(cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`)
+	# Install the desired version of Node
+	source ~/.nvm/nvm.sh
+	nvm install $TRAVIS_NODE_VERSION > /dev/null
+	npm install -g > /dev/null
+	npm install > /dev/null
+	set -x
+	stylelint --config ../.stylelintrc "styles/prosilver/theme/*.css"
+fi
```
Comment (hanakin) [inline travis/check-stylesheet.sh]: probably need to lint the admin css as well to ensure complete standards

Comment (marc1706) [inline travis/check-stylesheet.sh]: Yep, will add the admin css as well.

Comment (marc1706) [inline travis/check-stylesheet.sh]: Added the admin CSS but that line is commented now due to the large amount of issues that are noted. I think it's best to fix these issues in a separate PR.

Comment (hanakin) [inline travis/check-stylesheet.sh]: agree also since the I still need to update the linting rules file

**Pull request merged**